akka-http-core/src/main/scala/akka/http/impl/engine/client/PoolGateway.scala

```diff
@@ -25,6 +25,15 @@ import scala.concurrent.{ Future, Promise }
  */
 private[http] final class PoolGateway(gatewayRef: ActorRef, val hcps: HostConnectionPoolSetup, val gatewayId: GatewayIdentifier)(implicit fm: Materializer) {
 
+  private val shutdownCompletedPromise: Promise[Done] = Promise[Done]()
+
+  {
+    implicit val ec = fm.executionContext
+    shutdownCompletedPromise.future.foreach { _ ⇒
+      gatewayRef ! Shutdown(this, shutdownCompletedPromise)
```

Comment:
On second though, this one doesn't seem correct: we already have the protocol to only complete the promise when the gateway actor has shutdown itself by passing in the promise which doesn't make sense any more now that the promise is complete first.